services-worker/src/main/java/edu/unc/lib/dl/cdr/services/processing/ApplyEnhancementServicesJob.java

```diff
@@ -22,70 +25,101 @@
 	private long recoverableDelay = 0;
 	private final EnhancementMessage message;
 	private ActivityMetricsClient metricsClient;
-	
-	
+	private ManagementClient fedoraManagementClient;
+
 	public ApplyEnhancementServicesJob(String pidString, boolean force) {
 		this.message = new EnhancementMessage(pidString, servicesMessageNamespace,
 				ServicesActions.APPLY_SERVICE_STACK.getName());
 		this.message.setCompletedServices(new ArrayList<String>());
 		this.message.setForce(force);
 	}
-	
+
 	public ApplyEnhancementServicesJob(String messagePid, String messageNamespace, String messageAction, String messageServiceName, List<String> filteredServices) {
 		this.message = new EnhancementMessage(messagePid, messageNamespace, messageAction, messageServiceName);
 		this.message.setFilteredServices(filteredServices);
 		this.message.setCompletedServices(new ArrayList<String>());
 	}
-	
+
 	public void setServices(List<ObjectEnhancementService> services) {
 		this.services = services;
 	}
-	
+
 	public void setRecoverableDelay(long recoverableDelay) {
 		this.recoverableDelay = recoverableDelay;
 	}
-	
+
 	public void setMetricsClient(ActivityMetricsClient operationMetricsClient) {
 		this.metricsClient = operationMetricsClient;
 	}
 
+	public void setFedoraManagementClient(ManagementClient fedoraManagementClient) {
+		this.fedoraManagementClient = fedoraManagementClient;
+	}
+
 	@Override
 	public void run() {
+		long backoffDelay = 10000;
+		int backoffAttempts = 1;
+		int maxBackoffAttempts = 5;
+
 		for (ObjectEnhancementService service : services) {
 			if (message.getFilteredServices() != null
 					&& !message.getFilteredServices().contains(service.getClass().getName())) {
 				continue;
 			}
-			
+
 			try {
 				if (!service.isApplicable(message)) {
 					continue;
 				}
 			} catch (EnhancementException e) {
 				LOG.error("Error determining applicability for service " + service.getClass().getName() + " and object " + message.getTargetID(), e);
 			}
-			
-			try {
-				applyService(service);
-				metricsClient.incrFinishedEnhancement(service.getClass().getName());
-			} catch (EnhancementException e) {
-				LOG.error("Error applying service " + service.getClass().getName() + " to object " + message.getTargetID(), e);
-				metricsClient.incrFailedEnhancement(service.getClass().getName());
-			} catch (Throwable t) {
-				metricsClient.incrFailedEnhancement(service.getClass().getName());
-				throw t;
+
+			while (backoffAttempts <= maxBackoffAttempts) {
```

Comment:
While looking at this, I started wondering if it'd make sense to have this retry forever when Fedora is not available or throws an authentication/WebServiceIOException exception.  The reason being that the failure isn't an issue with the job, but with the environment, so skipping over the job won't help.  What do you think?  Would it make sense to maybe increase the delay and make it retry forever in the two cases I mentioned?

For startup of the work daemon i think its fine for it to have max retries before giving up since nothing from the queue would be lost.

It'd be kind of nice to get notifications of some sort when these things are failing, but I don't really want to over engineer this since it's not going to be used for that long.  So lets hold off on that until the need comes up.